### PR TITLE
Document date math use in the rollover API

### PR DESCRIPTION
**Pull request by clintongormley**
Relates to #20709

### CODE REVIEW

**Comment by clintongormley** [comment]
@s1monw could you take a look please?

File: docs/reference/indices/rollover-index.asciidoc [hunk 1]
```diff
@@ -84,6 +84,58 @@ POST /my_alias/_rollover/my_new_index_name
...
+{
+  "conditions": {
+    "max_age":   "1d"
+  }
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[skip:Requires 24h to pass]
+<1> Creates an index named (e.g.) `logs-2016.10.31`
+<2> Rolls over to a new index with today's date, e.g. `logs-2016.11.01`
```
Comment (s1monw) [inline docs/reference/indices/rollover-index.asciidoc]: sorry this is not how it works you have to create an index `<logs-{now/d}-1>` first and then when you rollover we increment the counter if it's a new day or not. So I think this entire writeup is wrong ?

**Review by s1monw — CHANGES_REQUESTED** [review]

File: docs/reference/indices/rollover-index.asciidoc (hunk 1 above)
Comment (clintongormley) [inline docs/reference/indices/rollover-index.asciidoc]: Doh

**Comment by clintongormley** [comment]
Docs fixed to represent what actually happens in the code :)

File: docs/reference/indices/rollover-index.asciidoc [hunk 2]
```diff
@@ -84,6 +84,71 @@ POST /my_alias/_rollover/my_new_index_name
...
 
 [float]
+=== Using date math with the rolllover API
+
+It can be useful to use <<date-math-index-names,date math>> to name the
+rollover index according to the date that the index rolled over, e.g.
+`logstash-2016.02.03`.  The rollover API supports date math, but requires the
+index name to end with a dash followed by a number, e.g.
+`logstash-2016.02.03-1` which is incremented every time the index is rolled
+over. For instance:
```
Comment (s1monw) [inline docs/reference/indices/rollover-index.asciidoc]: should we also add an example how to search those indices then?

**Review by s1monw — APPROVED** [review]
LGTM left one minor comment

File: docs/reference/indices/rollover-index.asciidoc (hunk 2 above)
Comment (clintongormley) [inline docs/reference/indices/rollover-index.asciidoc]: Example added

**Pull request merged**